client/errors.go

```diff
@@ -35,7 +35,7 @@ func ErrorConnectionFailed(host string) error {
 
 type notFound interface {
 	error
-	NotFound() bool // Is the error a NotFound error
+	NotFound()
```

Comment:
Actually, thinking of reverting this change; I already updated the `IsNotFound()` utility to match both this error, and the `errdefs.NotFound()` interface, so perhaps we should keep this one (for now) to make both the old and new ones work.

@cpuguy83 wdyt?